eval/context.go

```diff
@@ -203,15 +220,41 @@ func (c *Context) HCLContext() *hcl.EvalContext {
 }
 
 // updateFunctions recreates the listed functions with latest evaluation context.
-func updateFunctions(ctx *Context) {
-	if len(ctx.profiles) > 0 {
-		jwtfn := lib.NewJwtSignFunction(ctx.profiles, ctx.eval)
-		ctx.eval.Functions[lib.FnJWTSign] = jwtfn
+func (c *Context) updateFunctions() {
```

Comment:
Does oauth2 functions depending on `ctx.eval` like `JwtSignFunction` ? Seems to be obsolete recreate those ones every time? (maybe saml too). Inline Attrs gets referenced just from the AccessControl side with current context. We should prefer splitting setup and update functions.